src/content_sao.cpp

```diff
@@ -1020,7 +1020,7 @@ void PlayerSAO::step(float dtime, bool send_recommended)
 		const ContentFeatures &c = m_env->getGameDef()->ndef()->get(n);
 		// If player is alive & no drowning & not in ignore, breathe
 		if (m_breath < m_prop.breath_max &&
-				c.drowning == 0 && n.getContent() != CONTENT_IGNORE && m_hp > 0)
+				(isImmortal() || c.drowning == 0) && n.getContent() != CONTENT_IGNORE && m_hp > 0)
```

Comment:
This is a bit weird. I think breath values should freeze like the HP as soon the player becomes immortal. Mods can still re-fill the breath if they want to restore the player's mortality.

Reply:
I agree. I'll remove this.

Reply:
Here's the modified condition:

```cpp
if (m_breath < m_prop.breath_max && c.drowning == 0 &&
		n.getContent() != CONTENT_IGNORE && m_hp > 0 && !isImmortal())
	setBreath(m_breath + 1);
```

Player "breathes" only if they're _not_ immortal.